obspy/conftest.py

```diff
@@ -237,17 +238,26 @@ def pytest_json_modifyreport(json_report):
 
 def get_dependency_info():
     """Add version info about ObsPy's dependencies."""
-    import pkg_resources
-    distribution = pkg_resources.get_distribution('obspy')
     version_info = {'obspy': obspy.__version__}
-    for req in distribution.requires():
-        name = req.name
-        version = pkg_resources.get_distribution(name).version
+    for req in importlib.metadata.requires("obspy"):
+        # can not find proper documentation for how exactly these requirement
+        # strings are defined, but it looks like a semicolon is used as a
+        # separator for additional definitions like "extra" in optional
+        # dependencies and old pkg_resources ignored those, so skip them with
+        # importlib too
```

Comment:
I think it follows [PEP508](https://peps.python.org/pep-0508/); you could [parse it with `packaging`](https://packaging.pypa.io/en/stable/requirements.html).